src/server/game/Chat/Chat.h

```diff
@@ -127,8 +129,10 @@ class TC_GAME_API ChatHandler
         GameTele const* extractGameTeleFromLink(char* text);
         bool GetPlayerGroupAndGUIDByName(const char* cname, Player*& player, Group*& group, ObjectGuid& guid, bool offline = false);
         std::string extractPlayerNameFromLink(char* text);
+        std::string extractPlayerNameFromLink(std::string const &text);
         // select by arg (name/link) or in-game selection online/offline player or self if a creature is selected
         bool extractPlayerTarget(char* args, Player** player, ObjectGuid* player_guid = nullptr, std::string* player_name = nullptr);
+        bool extractPlayerTarget(std::string const &nameStr, Player** player, ObjectGuid* player_guid = nullptr, std::string* player_name = nullptr);
```

Comment:
Don't touch this class to add std::string overloads (will be ported from 3.3.5 later)

Reply:
I didn't really understand what you mean here.
Is the problem with overloads? That is, if I rename the methods is that fine? or should I not create any methods with std::string in the header?

Reply:
This has to do with significant improvements (black-magic like) done to command argument handling done in the other branch (you would be adding code that will definitely be unused later)

Another thing are the issues I mentioned in other comments

(all of current, "old" handling is based on `strtok` and relies on having calls to  `strtok` done)

Reply:
Right I saw all the strtok handling, I thought making some new methods with std::string and tokenizer is gonna make it better.

I'm still not sure what I can use though; can I just move my methods into the .cpp file and make them non-member methods? is that fine?